### DISCUSSION TITLE
Retry failed tests without overwriting reports

### DISCUSSION THREAD

**Discussion by JacobHearst** [fastlane/fastlane#19109] [Q&A]
Hi all,
I'm working on a project that has a tool which (after testing) reads and uploads the junit report to a TestRail instance. Our project has some flaky UI tests, so to help stabilize our builds, I added the `number_of_retries` option to our `scan` call. It works just as expected, _except_ when it retries, it overwrites the junit report from the previous run. This is problematic because it means our TestRail uploader will only see a single test result and our reports will say that only a single test passed.

Is there a way to retry tests without overwriting the existing report? Overwriting the report essentially makes the whole reporting feature worthless since you're only getting information from (at worst) a single test. Ideally I'd like to have fastlane retry failed tests and replace the failed results in the existing report with the result from retrying but even separate files would be better than just overwriting.

**Comment by ChristopherDrum** [comment]
I'm also interested in this. Because the report gets overwritten, it destroys my ability to collect unit test time metrics on my CI system.
